Initial upgrade guide v8-v9 project (#21012)
microsoft/fluentui: apps/upgrade-v8-v9/README.md

> # React Fluent UI v8 to v9 Upgrade Guide

Reviewer: ```suggestion
# Fluent UI React v8 to v9 Upgrade Guide
```

> 

Author: I'll follow up with Paul on the naming, but currently we call it v8 and v9 other places.

## How to use this guide

> This document will help guide you through building a plan to upgrade from React Fluent UI v8 to v9.

Reviewer: ```suggestion
This document will help guide you through building a plan to upgrade from `@fluentui/react` version 8 to `@fluentui/react-components` version 9.
```

- Before: v8 components only
- During: a mix of v8 and v9 components
- After: v9 components only

As you upgrade each component, leverage the accompanying web application.
It includes both v8 examples and matching v9 usage examples.
By building v8-only, mixed, or v9-only files of the application you can compare the application between phases.
The application was designed to let you run code-mods against it to learn about the kinds of upgrade changes.

You may notice the examples are pulled from storybook, but this app does not use storybook. Reasons:

- This isn't meant to replace the v8 or v9 storybooks, so we don't need any of the storybook features.
- The application is built with only fluent and react dependencies so you can inspect the resulting bundle size. Storybook was too big a dependency to add.
- We wanted to leverage the create-react-app fluent template as the starting point.Storybook is designed to be used during development and isn't the default production compile path of the product.
- We only want the minimal set of usage examples specific to upgrade scenarios. Storybook is built to provide multiple variations of component usage.

## Motivation

> React Fluent UI v9 provides a signficantly improved set of components, composition and deployment features, and a modern design language for styling and theming.

Reviewer: ```suggestion
Fluent UI React v9 provides a significantly improved set of components, composition and deployment features, and a modern design language for styling and theming.
```
